tests/test_loss_functions.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+import torch
+
+from modalities.batch import InferenceResultBatch
+from modalities.loss_functions import AsymmNCELoss
+
+
+@pytest.fixture
+def dummy_result_batch() -> InferenceResultBatch:
+    predictions = {"embedding": torch.FloatTensor(1024, 512).uniform_(-50, 50)}
+    targets = {"target": torch.zeros(1024, 512)}
+    batch_dim = 1024
+    result_batch = InferenceResultBatch(targets, predictions, batch_dim)
+    return result_batch
+
+
+# calculating asymmetric NCELoss between a batch of embeddings and itself --> zero
+@pytest.mark.parametrize("key", ["embedding"])
+def test_AsymmNCELoss_is_zero(dummy_result_batch, key):
```

Comment:
what about adding another example for which we calculate the correct score manually and then we check this in a test case as well? Right now we only check for a batch of two identical embedding matrices.

Reply:
A new test is added to check the correctness of nce_loss function. 

Update to nce_loss:
as mentioned above the code was adapted from MIL-NCE loss from (https://arxiv.org/pdf/1912.06430.pdf). however, since the goal is to use this loss for training CoCa model, the code is now adapted to reflect the contrastive loss presented in CoCa paper (https://arxiv.org/pdf/2205.01917.pdf). There is still an optional variable for calculating the loss only w.r.t. one modality or in both directions. For CoCa model a bidirectional loss should be applied. 

Please could you review the changes and let me know of any other updates. Hand-written calculations of the loss examples are available in case you are interested. 

Many thanks for your review!